modules/core/src/v2/coins/eos.ts

```diff
@@ -408,4 +438,268 @@ export class Eos extends BaseCoin {
+        }
+        if (source === 'backup' && !isKrsRecovery) {
+          return HDNode.fromBase58(sjcl.decrypt(passphrase, key));
+        }
+      } catch (e) {
+        throw new Error(`Failed to decrypt ${source} key with passcode - try again!`);
+      }
+    }
+    try {
+      return HDNode.fromBase58(key);
+    } catch (e) {
+      throw new Error(`Failed to validate ${source} key - try again!`);
+    }
+  };
+
+  /**
+   * Prepare and validate all keychains from the keycard for recovery
+   * @param userKey
+   * @param backupKey
+   * @param recoveryDestination
+   * @param krsProvider
+   * @param walletPassphrase
+   */
+  initiateRecovery({
+    userKey,
+    backupKey,
+    recoveryDestination,
+    krsProvider,
+    walletPassphrase,
+  }: BaseCoinInitiateRecoveryOptions): Bluebird<HDNode[]> {
+    return co(function*() {
+      const isKrsRecovery = backupKey.startsWith('xpub') && !userKey.startsWith('xpub');
+      const isUnsignedSweep = backupKey.startsWith('xpub') && userKey.startsWith('xpub');
+
+      if (isKrsRecovery) {
+        const krsProviderConfig = config.krsProviders[krsProvider];
+        if (_.isUndefined(krsProviderConfig)) {
+          throw new Error('unknown key recovery service provider');
+        }
+
+        if (!krsProviderConfig.supportedCoins.includes(this.getFamily())) {
+          throw new Error('specified key recovery service does not support recoveries for this coin');
+        }
+      }
+
+      const keys = [
+        this.validateKey({
+          key: userKey,
+          source: 'user',
+          passphrase: walletPassphrase,
+          isKrsRecovery,
+          isUnsignedSweep,
+        }),
+        this.validateKey({
+          key: backupKey,
+          source: 'backup',
+          passphrase: walletPassphrase,
+          isKrsRecovery,
+          isUnsignedSweep,
+        }),
+      ];
+      if (!this.isValidAddress(recoveryDestination)) {
+        throw new Error('Invalid destination address!');
+      }
+
+      return keys;
+    }).call(this);
+  }
+
+  getAccountFromNode(address): Bluebird<any> {
+    return co(function*() {
+      const nodeUrl = this.getPublicNodeUrl();
+      const result = yield this.bitgo.post(nodeUrl + '/v1/chain/get_account').send({ account_name: address });
+      if (!result || result.status !== 200) {
+        throw new Error('unable to fetch account from node: ' + nodeUrl);
+      }
+      return result.body;
+    }).call(this);
+  }
+
+  getTransactionHeadersFromNode(): Bluebird<any> {
+    return co(function*() {
+      const nodeUrl = this.getPublicNodeUrl();
+      const chainInfoResult = yield this.bitgo.post(nodeUrl + '/v1/chain/get_info').send();
+      if (!chainInfoResult || chainInfoResult.status !== 200) {
+        throw new Error('unable to fetch chain info from node: ' + nodeUrl);
+      }
+      const chainInfo = chainInfoResult.body;
+
+      const headBlockInfoResult = yield this.bitgo.post(nodeUrl + '/v1/chain/get_block').send({
```

Comment:
Since you're already using a public node (i.e. assuming that bitgo is down), can we just use the get_block EOS api to get this info?

Reply:
oh man yep my bad.. thought this was hitting the ims